modules/dkan/dkan_datastore/modules/dkan_datastore_api/dkan_datastore_api.module

```diff
@@ -1123,3 +1123,180 @@ function dkan_datastore_api_debug($data_select) {
   $sql = str_replace('\n', ' ', $sql);
   return (string) $sql;
 }
+
+/**
+ * Implements hook_services_resources_alter().
+ */
+function dkan_datastore_api_services_resources_alter(&$resources, $endpoint) {
+  if (isset($resources['node']['operations']['create'])) {
+    $resources['node']['operations']['create']['callback'] =
+      "_dkan_datastore_api_resource_create";
+  }
+
+  if (isset($resources['node']['operations']['update'])) {
+    $resources['node']['operations']['update']['callback'] =
+      "_dkan_datastore_api_resource_update";
+  }
+}
+
+/**
+ * Services operation callback: node creation.
+ */
+function _dkan_datastore_api_resource_create($request) {
+  $node_array = _services_arg_value($request, 'node');
+  $node = (object) $node_array;
+
+  if (isset($node->type)) {
+    if ($node->type == "dataset") {
+
+      try {
+        field_attach_validate('node', $node);
+        _dkan_datastore_api_field_validation($node);
+      }
+      catch (\Exception $e) {
+        return services_error($e->getMessage());
+      }
+
+      node_save($node);
+
+      $nid = $node->nid;
+      $response = array('nid' => $nid);
+      if ($uri = services_resource_uri(array('node', $nid))) {
+        $response['uri'] = $uri;
+      }
+    }
+    else {
+      $response = _node_resource_create($request);
+    }
+  }
+  else {
+    return services_error("Type not set for node");
+  }
+
+  return $response;
+}
+
+/**
+ * Services operation callback: node update.
+ */
+function _dkan_datastore_api_resource_update($nid, $request) {
```

Comment:
No access control? Also does this create a new revision?

Reply:
The code is working with the entity api directly so it will do whatever it does. I believe if no vid is set it will create a new version. I am not sure what the behavior is if a vid is set.